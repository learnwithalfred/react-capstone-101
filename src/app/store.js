import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from '../features/rockets/rocketSlice';
import missionReducer from '../features/missions/missionsSlice';
import themeReducer from '../features/theme/themeSlice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
    theme: themeReducer,
  },
});

export default store;
