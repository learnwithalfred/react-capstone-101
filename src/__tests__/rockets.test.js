import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Rockets from '../features/rockets/rockets';
import store from '../app/store';

describe('Rockets should render correctly', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('container should have first child', () => {
    const { container } = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(container.firstChild).toBeTruthy();
  });

  it('table should have a h2 header', () => {
    const { container } = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(container.querySelector('h2')).toBeTruthy();
  });
});
