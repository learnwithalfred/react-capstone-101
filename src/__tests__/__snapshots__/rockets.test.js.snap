// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets should render correctly should render correctly 1`] = `
<div>
  <h2>
    Loading...
  </h2>
</div>
`;
